### PR TITLE
Cleanup log filter code, and add supports for LT and GT filters.

### PR DESCRIPTION
**Pull request by zwoop**
The motivation for the LT/GT operators is to allow custom log formats to do more complex filters, even replace the slow log functionality. One could do a filter now that checks if the "cache open read" is slower than a certain number of ms, e.g.

```
  filters:
  - name: slowfilter
    action: accept
    condition: %<{TS_MILESTONE_CACHE_OPEN_READ_END-TS_MILESTONE_CACHE_OPEN_READ_BEGIN}msdms> GT 10
```

The cleanup was because when I wanted to add this, I realized this code was inefficient, causing double evaluation of the filters including calls to malloc.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci autest]

File: include/proxy/logging/LogFilter.h [hunk 1]
```diff
@@ -134,12 +143,34 @@ class LogFilterString : public LogFilter
...
 
+  // This assumes that s1 is all uppercase, hence we hide this in here specifically
+  static const char *
+  strstrcase(const char *s0, const char *s1)
+  {
+    while (*s0) {
+      const char *h = s0;
+      const char *n = s1;
+
+      while (*n && (std::toupper((unsigned char)*h) == *n)) {
```
Comment (ywkaras) [inline include/proxy/logging/LogFilter.h:159]: Casting like it's 1999.

File: src/proxy/logging/LogFilter.cc [hunk 2]
```diff
@@ -557,8 +614,14 @@ LogFilterInt::LogFilterInt(const char *name, LogField *field, LogFilter::Action
                                            // pointer in @c field.
 
   if (n) {
+    if (m_operator >= Operator::LT && m_operator <= Operator::GTE && n > 1) {
+      Fatal("Operator %s can only have one value", OPERATOR_NAME[m_operator]);
+      return;
+    }
+
```
Comment (ywkaras) [inline src/proxy/logging/LogFilter.cc:621]: As long as you're in here, the test for non-zero n on line 644 is redundant.

**Review by bryancall — CHANGES_REQUESTED** [review]

**Comment by bryancall** [comment]
[approve ci clang-analyzer]

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci centos]

**Review by bryancall — APPROVED** [review]

**Comment by zwoop** [comment]
[approve ci autest]

**Pull request merged**